frontend/src/component/ai/AIChat.tsx

```diff
@@ -88,22 +92,45 @@ export const AIChat = () => {
 
     const [messages, setMessages] = useState<ChatMessage[]>([]);
 
+    const isAtEndRef = useRef(true);
     const chatEndRef = useRef<HTMLDivElement | null>(null);
 
-    const scrollToEnd = (options?: ScrollIntoViewOptions) => {
+    const scrollToEnd = (options?: ScrollOptions) => {
         if (chatEndRef.current) {
-            chatEndRef.current.scrollIntoView(options);
+            const shouldScroll = !options?.onlyIfAtEnd || isAtEndRef.current;
+
+            if (shouldScroll) {
+                chatEndRef.current.scrollIntoView(options);
+            }
         }
     };
 
-    useEffect(() => {
-        scrollToEnd({ behavior: 'smooth' });
-    }, [messages]);
-
     useEffect(() => {
         scrollToEnd();
+
+        const intersectionObserver = new IntersectionObserver(
+            ([entry]) => {
+                isAtEndRef.current = entry.isIntersecting;
+            },
+            { threshold: 1.0 },
+        );
+
+        const target = chatEndRef.current;
+        if (target) {
+            intersectionObserver.observe(target);
+        }
+
+        return () => {
+            if (target) {
```

Comment:
Should this be like this? target will get a reference to the target and the function will just keep it referenced
```suggestion
            if (chatEndRef.current) {
```

Reply:
Good point! I simplified this to always use the ref: https://github.com/Unleash/unleash/pull/8478/commits/8d473e1ae504d8c8e1f7c1fb258e01a31897c464